### DISCUSSION TITLE
I think I run into a chrome bug?

### DISCUSSION THREAD

**Discussion by shi-yan** [gpuweb/gpuweb#4431] [Q&A] [answered]
I think I run into a chrome bug?

I have a compute shader like this:

```
@binding(0) @group(0) var<storage, read> centroid :array<vec3<f32>>;
    @binding(1) @group(0) var<storage, read_write> distance: array<f32>;
    @binding(2) @group(0) var<uniform> splatCount: u32;
    
    @compute @workgroup_size(256)
    fn main(@builtin(global_invocation_id) GlobalInvocationID : vec3<u32>,
      @builtin(local_invocation_id) LocalInvocationID: vec3<u32>,
      @builtin(workgroup_id) WorkgroupID: vec3<u32>) {
          if (GlobalInvocationID.x ==1) {
            var gid = GlobalInvocationID.x;
                var center:vec3<f32> = centroid[1];
                distance[0] = center.x;
                distance[1] = center.y;
                distance[2] = center.z;
          }
    }
```

the shader reads the value at the index 1 from an array of type `array<vec3<f32>>` and write to an output `array<f32>`

hence if the input `array<vec3<f32>>` is initialized by `new Float32Array([1,2,3,4,5,6,7,8,9])`,
the output array should be [4,5,6];

but the actual output is [5,6,7];

it seems that dawn thinks it was `array<vec4<f32>>` instead of `array<vec3<f32>>`

the minimal reproducible code:

https://raw.githubusercontent.com/shi-yan/WebGPUUnleashed/main/5_08_gaussian_splatting_5/bug.html

**Comment by shaoboyan** [comment, marked as answer]
I think this is due to padding rules : https://www.w3.org/TR/WGSL/#alignment-and-size.
Alignment of vec3 floats is 16 bytes instead of 12 bytes. So the initialized array should obey the padding rules.

**Reply by greggman** [reply]
Correct, `vec3<f32>` are aligned to 16 bytes

https://webgpufundamentals.org/webgpu/lessons/resources/wgsl-offset-computer.html#x=5d000001005800000000000000003d8888623728a306fc320e1a9b5df669b4aa9cbb003a1a3a0818b7274236b5a27c0c5d753eee27df84f54e745763ce49447aa83dea8791e515e7d40e3311e5aa8cf118b8f419338c930e4768086bc6ff513520a8903dbd3ffffe1ab800

**Reply by shi-yan** [reply]
Thank you very much!
